src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/GlobalSingleInstanceResponseTracker.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans.GrainDirectory;
+using Orleans.SystemTargetInterfaces;
+
+namespace Orleans.Runtime.GrainDirectory 
+{
+    /// <summary>
+    /// A class that encapsulates response processing logic.
+    /// It is a promise that fires once it has enough responses to make a determination.
+    /// </summary>
+    internal class GlobalSingleInstanceResponseTracker : TaskCompletionSource<GlobalSingleInstanceResponseTracker.Outcome> {
+
+        public enum Outcome {
+            Succeed,
+            RemoteOwner,
+            RemoteOwnerLikely,
+            Inconclusive
+        }
+
+        private readonly GrainId grain;
+        private RemoteClusterActivationResponse[] responses;
+
+        public AddressAndTag RemoteOwner;
+        public string RemoteOwnerCluster;
+
+        public GlobalSingleInstanceResponseTracker(RemoteClusterActivationResponse[] responses, GrainId grain)
+        {
+            this.responses = responses;
+            this.grain = grain;
+
+            Notify();
```

Comment:
`Notify` is a bad name. Will rename to `CheckIfDone`.

It's called in the constructor because if the MultiCluster contains only one cluster, then we are done immediately, without waiting for any responses.